README.md

```diff
@@ -12,76 +12,143 @@ Follow these steps to set up the project:
-            }],
-            "boolean": [...]
-          },
-          "methods": [{
-            "name": "method1",
-            "accessModifier": "private",
-            "parameters": {"param1": "String", "param2": "int"},
-            "returnType": "void"
-          }],
-          "constructors: [{
-            "name": "ex1",
-            "accessModifier": "public",
-            "parameters": {"param1": "String", "param2": "int"},
-            "returnType": null
-          }]
-    },
-    {...},
-    {...}
+              "access_modifier": "public",
+            },
+            {
+              "name": "bar",
+              "type": "int",
+              "access_modifier": "private"
+            }
+          ],
+          "methods": [
+            {
+              "name": "getBar",
+              "access_modifier": "public",
+              "parameters": [],
+              "return_type": "int"
+            },
+            {
+              "name": "setFoo",
+              "access_modifier": "public",
+              "parameters": [
+                {
+                  "name": "foo",
+                  "type": "String"
+                }
+              ],
+              "return_type": "void"
+            }
+          ],
+          "constructors": [
+            {
+              "name": "C1",
+              "access_modifier": "public",
+              "parameters": [
+                {
+                  "name": "foo",
+                  "type": "String",
+                },
+                {
+                  "name": "bar",
+                  "type": "int"
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
   ]
 }
 ```
 
-### JavaClass Class
-Fields:
-- `name`: String
-- `type`: Enum (Interface, Abstract, Concrete, etc.)
-- `accessModifier`: Enum (private, public, protected)
-- `lineCount`: int (number of lines)
-- `imports`: list of String
-- `fields`: Map of type to list of JavaField Class (Easier to work with when rendering groups of fields according to type)
-- `methods`: list of JavaMethod Class
-- `constructors`: list of JavaMethod Class
-
-### JavaField Class
-Fields:
-- `name`: String
-- `type`: String
-- `accessModifier`: Enum (private, public, protected)
-
-### JavaMethod Class
-Fields:
-- `name`: String
-- `accessModifier`: Enum (private, public, protected)
-- `parameters`: map of: name to type (Not sure what we need here...)
-- `returnType`: String
-- `~~fields~~` (Add local fields as stretch goal)
+## Visualization of Java Project
+
+- [Project](#project-class) = Map of islands
+- [Package](#package-class) = ???
+- [Class](#class-class) = Island (size is determined by number of lines of code)
+  - `Concrete` = ???
+  - `Interface` = ???
+  - `Abstract` = ???
+  - `public` = ???
+  - `private` = ???
+  - `protected` = ???
+  - `import` = ???
+- [Field](#field-class) = ???
+- [Method](#method-class) = ???
+- [Constructor](#constructor-class) = ???
+- [Parameter](#parameter-class) = ???
```

Comment:
We have to think carefully about how to visualize differnet parts of Java code.

Reply:
Currently I'm using this for the island icons. Definitely not set in stone or anything... is there something that isn't covered though, or something that would work better as an icon? Extra information like parameters and names are in the tooltip bubble: 

![legendPNG](https://user-images.githubusercontent.com/48613270/83934539-c1e90c80-a766-11ea-875e-12a6d918753d.png)

I'm having a bit of a hard time figuring out what the above syntax means though. For example,  what does '#parameter-class' in [Parameter](#parameter-class) = ??? mean, and what does the ??? stand for?

Reply:
`???` means the visualization of the entity is not finalized yet. `#parameter-class` is a link to the title `Parameter Class`.

Reply:
Java has other primitive types like `long`, `double`, etc. Do we want to visualize them as ponds or something else?

Reply:
Sorry, I was thinking we could replace 'Object' (the rock)  with 'other'

and yeah, maybe we could depict the 8 primitive data types as an icon: https://www.w3schools.com/java/java_data_types.asp

Reply:
actually, do we need to have an icon for 'void'? What is it used for other than in method declarations (which would be shown through the tooltip text)?

Reply:
How about a unique icon for each of:
boolean type
numeric type
string
collection
map
other

based on this: https://www.geeksforgeeks.org/data-types-in-java/ and this: https://www.geeksforgeeks.org/collections-in-java-2/